ast_convenience.ml

```diff
@@ -28,6 +28,32 @@ module Label = struct
 
 end
 
+module Constant = struct 
+  type t = 
+     Pconst_integer of string * char option 
+   | Pconst_char of char 
+   | Pconst_string of string * string option 
+   | Pconst_float of string * char option 
+
+  let of_constant = function       
+    | Asttypes.Const_int32(i) -> Pconst_integer(Int32.to_string i, Some 'l')
+    | Asttypes.Const_int64(i) -> Pconst_integer(Int64.to_string i, Some 'L')
+    | Asttypes.Const_nativeint(i) -> Pconst_integer(Nativeint.to_string i, Some 'n')
+    | Asttypes.Const_int(i) -> Pconst_integer(string_of_int i, None)
+    | Asttypes.Const_char c -> Pconst_char c 
+    | Asttypes.Const_string(s, s_opt) -> Pconst_string(s, s_opt) 
+    | Asttypes.Const_float f -> Pconst_float(f, None)
+
+  let to_constant = function 
+    | Pconst_integer(s, Some 'l') -> Asttypes.Const_int32 (Int32.of_string s)
+    | Pconst_integer(s, Some 'L') -> Asttypes.Const_int64 (Int64.of_string s)
```

Comment:
@hhugo :+1:. Thank you for the pointer. Indeed 4.03 has more robust `int{32,64,nativeint}` conversion functionality. I have back-ported this to `ppx_tools` 4.02 branch.